### ISSUE TITLE
Swallowed / Nympho not matching

### ISSUE THREAD

**Issue by Spanishu** [PAhelper/PhoenixAdult.bundle#120]
I have issues lately with Nympho as well as with Swallowed.com 
It seems that the name gets too reduced?

```
2019-03-04 14:57:21,442 (70eff460) :  INFO (__init__:45) - *******MEDIA TITLE****** Swallowed.com Sarah Vandella proves
©2019-03-04 14:57:21,445 (70eff460) :  INFO (__init__:52) - Getting Search Settings for: Swallowed.com Sarah Vandella proves
2019-03-04 14:57:21,462 (70eff460) :  INFO (PAsearchSites:691) - ^^^^^^^ siteID: 282
2019-03-04 14:57:21,464 (70eff460) :  INFO (PAsearchSites:693) - ^^^^^^^ Shortening Title
2019-03-04 14:57:21,466 (70eff460) :  INFO (PAsearchSites:694) - swallowed vs swallowed
2019-03-04 14:57:21,468 (70eff460) :  INFO (PAsearchSites:697) - 1
2019-03-04 14:57:21,470 (70eff460) :  INFO (PAsearchSites:709) - 5
2019-03-04 14:57:21,472 (70eff460) :  INFO (PAsearchSites:744) - 6
2019-03-04 14:57:21,473 (70eff460) :  INFO (PAsearchSites:744) - searchTitle (before date processing): Sarah Vandella proves
2019-03-04 14:57:21,475 (70eff460) :  INFO (__init__:62) - Search Title: Sarah Vandella proves
2019-03-04 14:57:21,477 (70eff460) :  INFO (__init__:71) - Sarah%20Vandella%20proves
2019-03-04 14:57:21,502 (70eff460) :  DEBUG (networking:161) - Fetching 'https://tour.swallowed.com/search/SaRaH' from the HTTP cache
2019-03-04 14:57:21,525 (70eff460) :  INFO (networkSteppedUp:18) - Matches found: 0
2019-03-04 14:57:21,540 (70eff460) :  DEBUG (networking:161) - Fetching 'https://tour.swallowed.com/search/SARAH' from the HTTP cache
2019-03-04 14:57:21,563 (70eff460) :  INFO (networkSteppedUp:21) - Matches found: 0
2019-03-04 14:57:21,578 (70eff460) :  DEBUG (networking:161) - Fetching 'https://tour.swallowed.com/search/sarah' from the HTTP cache
```

**Comment by PAhelper** [comment]
Yeah I've been struggling with this too. I'll keep fiddling with it and see if I can't get better results... The search bar in the original site is pretty finicky.

**Comment by PAhelper** [comment]
I've changed the search back to just searching whatever you type into the Match field of Plex... See if you get better results now.

**Comment by Spanishu** [comment]
sorry for late reply but I have just tried with the latest release and this is the result `2019-03-15 10:40:29,438 (70eff460) :  INFO (__init__:45) - *******MEDIA TITLE****** nympho bunny hops on cock
2019-03-15 10:40:29,440 (70eff460) :  INFO (__init__:52) - Getting Search Settings for: nympho bunny hops on cock
2019-03-15 10:40:29,442 (70eff460) :  INFO (PAsearchSites:676) - mediaTitle w/ possible abbrieviation: nympho bunny hops on cock
2019-03-15 10:40:29,448 (70eff460) :  INFO (PAsearchSites:764) - mediaTitle w/ possible abbrieviation fixed: nympho bunny hops on cock
2019-03-15 10:40:29,464 (70eff460) :  INFO (PAsearchSites:779) - ^^^^^^^ siteID: 284
2019-03-15 10:40:29,466 (70eff460) :  INFO (PAsearchSites:781) - ^^^^^^^ Shortening Title
2019-03-15 10:40:29,468 (70eff460) :  INFO (PAsearchSites:782) - nympho vs nympho
2019-03-15 10:40:29,470 (70eff460) :  INFO (PAsearchSites:785) - 1
2019-03-15 10:40:29,472 (70eff460) :  INFO (PAsearchSites:828) - 6
2019-03-15 10:40:29,474 (70eff460) :  INFO (PAsearchSites:828) - searchTitle (before date processing): bunny hops on cock
2019-03-15 10:40:29,476 (70eff460) :  INFO (__init__:62) - Search Title: bunny hops on cock
2019-03-15 10:40:29,478 (70eff460) :  INFO (__init__:71) - bunny%20hops%20on%20cock
2019-03-15 10:40:29,496 (70eff460) :  DEBUG (networking:166) - Requesting 'https://tour.nympho.com/search/bunny%20hops%20on%20cock'
2019-03-15 10:40:30,778 (70eff460) :  INFO (networkSteppedUp:54) - Matches found: 0
2019-03-15 10:40:30,806 (70eff460) :  DEBUG (runtime:88) - Sending packed state data (992 bytes)
2019-03-15 10:40:30,808 (70eff460) :  DEBUG (runtime:924) - Response: [200] str, 320 bytes`

Bizarre is if i take the link [https://tour.nympho.com/search/bunny%20hops%20on%20cock](url) it finds it perfectly...

**Comment by PAhelper** [comment]
Yeah that's what I was battling when I was truncating the search results and doing weird stuff with the capitalization. In my testing, it seemed to work better in CAPS sometimes, others times in title case, etc. I still haven't worked out exactly how to get these searches to work right all the time.

**Comment by Spanishu** [comment]
One thing I noted was clearing the cache help to get at least one result.. Maybe that is another thing to look at.

**Comment by Spanishu** [comment]
Just tried on Swallowed with Rosalyn is at it again. The log shows something interesting. The site would expect the spaces to be filled with - and not with %

`2019-03-22 13:43:56,491 (70eff460) :  INFO (PAsearchSites:871) - searchTitle (before date processing): Rosalyn is at it again
2019-03-22 13:43:56,493 (70eff460) :  INFO (__init__:62) - Search Title: Rosalyn is at it again
2019-03-22 13:43:56,495 (70eff460) :  INFO (__init__:71) - Rosalyn%20is%20at%20it%20again
2019-03-22 13:43:56,513 (70eff460) :  DEBUG (networking:166) - Requesting 'https://tour.swallowed.com/search/Rosalyn%20is%20at%20it%20again'
2019-03-22 13:43:57,948 (70eff460) :  INFO (networkSteppedUp:8) - Matches found: 0
2019-03-22 13:43:57,957 (70eff460) :  DEBUG (networking:166) - Requesting 'https://tour.swallowed.com/search/ROSALYN IS AT IT AGAIN'
2019-03-22 13:43:58,887 (70eff460) :  INFO (networkSteppedUp:11) - Matches found: 0
2019-03-22 13:43:58,897 (70eff460) :  DEBUG (networking:166) - Requesting 'https://tour.swallowed.com/search/rosalyn is at it again'
2019-03-22 13:43:59,874 (70eff460) :  INFO (networkSteppedUp:14) - Matches found: 0
2019-03-22 13:43:59,939 (70eff460) :  DEBUG (runtime:88) - Sending packed state data (8460 bytes)
2019-03-22 13:43:59,941 (70eff460) :  DEBUG (runtime:924) - Response: [200] str, 320 bytes
`

maybe that helps?

**Comment by d2dyno** [comment]
Many issues on my side too. Even searching the fully correct title won't work.

One thing I did notice, they have an autocomplete search that fully works, but hitting enter their search is abysmal.

`<input type="text" class="search-field" id="search" placeholder="Search Swallowed" name="q" value="ORAL ENTERTAINMENT WITH PENELOPE AND GIA" autocomplete="off" data-ajax-search-url="https://tour.swallowed.com/search-preview">`

**Comment by chonka1** [comment]
Anything new  on this?  I've been using the exact titles with %20 in the spaces to make it work, but would be nice if it just searched it like the autocomplete.

**Comment by Spanishu** [comment]
would it be possible to make a url match like we have in passion-hd? Not optimal but clearly better than nothing.

**Comment by d2dyno** [comment]
@claygoldfinch want to work your magic on this?

**Comment by Spanishu** [comment]
@claygoldfinch any chance that you could do your magic? I have quite a bit of content from those sites and they are the only ones not working. Any support much appreciated !

**Comment by claygoldfinch** [comment]
Been playing around with the search function without much luck. Direct URL match may be the way to go.

**Comment by kamuk90** [comment]
match url for nympho in networkSteepedUp.py file
```
def searchNympho(results,encodedTitle,title,searchTitle,siteNum,lang,searchByDateActor,searchDate, searchSiteID):
    if searchSiteID != 9999:
        siteNum = searchSiteID
    searchString = searchTitle.replace(" ", "-").replace(",", "").replace("'", "").replace("?", "")
    Log("searchString: " + searchString)
    if "/" not in searchString:
        searchString = searchString.replace("-", "/", 1)
        Log("searchString formatted")
    searchResults = HTML.ElementFromURL(PAsearchSites.getSearchSearchURL(siteNum) + searchString)

    Log("Matches found: " + str(len(searchResults.xpath('//div[@class="container model-page-wrap"]'))))
    for searchResult in searchResults.xpath('//div[@class="container model-page-wrap"]'):
        titleNoFormatting = searchResult.xpath('//div[@class="content-meta"]/h2')[0].text_content().strip()
        curID = searchResult.xpath('//input[@id="copy-url"]')[0].get('value').replace('/','_').replace('?','!')
        releaseDate = parse(searchResult.xpath('//span[@class="date hidden-xs"]')[0].text_content().strip()).strftime('%Y-%m-%d')
        if searchDate:
            score = 100 - Util.LevenshteinDistance(searchDate, releaseDate)
        else:
            score = 100 - Util.LevenshteinDistance(searchTitle.lower(), titleNoFormatting.lower())
        Log("curID: " + curID)
        results.Append(MetadataSearchResult(id = curID + "|" + str(siteNum), name = titleNoFormatting + " ["+PAsearchSites.getSearchSiteName(siteNum)+"] " + releaseDate, score = score, lang = lang))
    return results
```

**Comment by Spanishu** [comment]
thanks a lot, will you put this into the workstream of PAHelper? Can we extend this to swallowed and true anal?

**Comment by claygoldfinch** [comment]
All sites fixed, PR #225.

@Spanishu 

(Thanks to @kamuk90 for his code)
